src/btree/bt_cursor.c

```diff
@@ -740,10 +740,22 @@ __wt_btcur_search_near(WT_CURSOR_BTREE *cbt, int *exactp)
                 goto done;
         }
 
+        /*
+         * It is not necessary to go backwards when search_near is used with a prefix, as cursor row
+         * search places us on the first key of the prefix range. All the entries before the key we
+         * were placed on will not match the prefix.
+         *
+         * For example, if we search with the prefix "b", the cursor will be positioned at the first
+         * key starting with "b". All the entries before this one will start with "a", hence not
+         * matching the prefix.
+         */
+        if (F_ISSET(cursor, WT_CURSTD_PREFIX_SEARCH))
+            goto done;
+
         /*
          * We walked to the end of the tree without finding a match. Walk backwards instead.
```

Comment:
@luke-pearson, question that is out of the scope of the PR but probably worth it:
Why do we walk backwards from the last position of the cursor? Why don't we save the initial position of the cursor and start from there and go backwards?

Reply:
That's a good question and I remember i brought it up with Keith B but I forget the actual answer, apparently its not that trivial to do that?

Reply:
I imagined it was not trivial, I will ask him and let you know!

Reply:
Oooh, I'd be interested to know as well if you do get an answer on that

Reply:
The answer is that the key could be gone by the time we restart the search, that's why we cannot do it.